.github/workflows/spread.yaml

```diff
@@ -0,0 +1,62 @@
+name: Spread
+on:
+  pull_request:
+  merge_group:
+  push:
+    branches:
+      - main
+
+jobs:
+  snap-build:
+    runs-on: [self-hosted, amd64]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build snap
+        uses: snapcore/action-build@v1
+        id: snap
+      - name: Upload snap artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: snap
+          path: ${{ steps.snap.outputs.snap }}
+
+  snap-tests:
+    runs-on: spread-installed
+    needs: [snap-build]
+    strategy:
+      fail-fast: false
+      matrix:
+        spread:
+          - "lxd:"
```

Comment:
This is not running anything at the moment by the way (in case you weren't aware)

Reply:
Thanks. This whole workflow doesn't run yet as I need spread-available runners to be added to craft-application